lib/Models/Terria.ts

```diff
@@ -1006,6 +1031,63 @@ export default class Terria {
           .query("")
           .hash("")
       );
+
+      if (!this.appBaseHref.endsWith("/")) {
+        console.warn(
+          `Terria expected appBaseHref to end with a "/" but appBaseHref is "${this.appBaseHref}". Routes may not work as intended. To fix this, try setting the "--baseHref" parameter to a URL with a trailing slash while building your map, or constructing the Terria object with an appropriate appBaseHref (with trailing slash).`
+        );
+      }
+
+      // /catalog/ and /story/ routes
+      if (newUrl.startsWith(this.appBaseHref)) {
+        const pageUrl = new URL(newUrl);
+        // Find relative path from baseURI to documentURI excluding query and hash
+        // then split into url segments
+        // e.g. "http://ci.terria.io/main/story/1#map=2d" -> ["story", "1"]
+        const segments = (pageUrl.origin + pageUrl.pathname)
+          .slice(this.appBaseHref.length)
+          .split("/");
+        if (
+          segments.length === 2 &&
+          segments[0] === "catalog" &&
+          segments[1].length > 0
+        ) {
```

Comment:
One small problem is that the test fails if I have a `/` after `catalog` or `story`. Should we be a bit more permissive?

Reply:
Maybe `/catalog/` should be a route that just opens the catalog, but `/story/` should not be a route IMO. It should only support `/story/:story-id`.

Reply:
Oh sorry, I meant `/` at the end like in `/story/foo/`

Reply:
Yeah, this is probably a requirement at least for story routes, since I think they're intended to be sometimes typed out by humans, not just linked to from media releases etc.